test/integration/users.test.js

```diff
@@ -0,0 +1,156 @@
+    let response = {};
+
+    beforeEach(async done => {
+      response = await supertest(app)
+        .post(postUsersPath)
+        .set(...defaultHeaders)
+        .send(usersTestData.createUserRequestBody);
+
+      done();
+    });
+
+    test('Should return a created status code', () => {
+      const { status } = response;
+
+      expect(status).toBe(201);
+    });
+
+    test('Should return the created user info without password', () => {
+      const { body } = response;
+      const userInfo = dbHelper.cleanRecord(body, true);
+
+      expect(userInfo).toMatchObject(usersTestData.expectedUserInfo);
+    });
+
+    test('Should save the user with the correct information', async done => {
+      const {
+        body: { id }
+      } = response;
+
+      const foundUser = await dbHelper.findUserById(id);
+      const cleanedFoundUser = dbHelper.cleanRecord(foundUser);
+
+      delete cleanedFoundUser.password;
+
+      expect(cleanedFoundUser).toStrictEqual({ id, ...usersTestData.expectedUserInfo });
+      done();
+    });
+  });
+
+  describe('Should fail correctly with schema error cases', () => {
+    const schemaErrorStatus = 422;
+
+    describe('Should throw an error if the password does not fullfil the schema', () => {
+      test('Should throw an error if the password length is lower than 8', async done => {
+        const shortPassword = '1234';
+
+        const { status, body } = await supertest(app)
+          .post(postUsersPath)
+          .set(...defaultHeaders)
+          .send({ ...usersTestData.createUserRequestBody, password: shortPassword });
+
+        expect(status).toBe(schemaErrorStatus);
+        expect(body).toStrictEqual(
+          usersTestData.getSchemaErrorResponse([
+            {
+              location: 'body',
+              msg: errorMessages.shortPassword,
+              param: 'password',
+              value: shortPassword
+            }
+          ])
+        );
+
+        done();
+      });
+
+      test('Should throw an error if the password is not alphanumeric', async done => {
+        const notAlphanumericPassword = '@#¢∞¬¬÷“”“∞';
+
+        const { status, body } = await supertest(app)
+          .post(postUsersPath)
+          .set(...defaultHeaders)
+          .send({ ...usersTestData.createUserRequestBody, password: notAlphanumericPassword });
+
+        expect(status).toBe(schemaErrorStatus);
+        expect(body).toStrictEqual(
+          usersTestData.getSchemaErrorResponse([
+            {
+              location: 'body',
+              msg: errorMessages.alphanumeric,
+              param: 'password',
+              value: notAlphanumericPassword
+            }
+          ])
+        );
+
+        done();
+      });
+    });
+
+    test('Should throw an error when an obligatory field is not provided', async done => {
+      const obligatoryFields = ['first_name', 'last_name', 'email', 'password'];
+
+      await Promise.all(
```

Comment:
Theses tests must be do in a better way, check `test.each` in this doc: https://jestjs.io/docs/api#testeachtablename-fn-timeout